Composer/packages/client/src/pages/botProject/PublishTargets.tsx

```diff
@@ -156,37 +141,56 @@ export const PublishTargets: React.FC<PublishTargetsProps> = (props) => {
     [publishTargets, projectId]
   );
 
-  useEffect(() => {
+  // add profile
+  const openAddProfileDialog = useCallback(() => {
     setDialogProps({
-      title: formatMessage('Add a publish profile'),
+      ...PublishProfileDialog.ADD_PROFILE,
       type: DialogType.normal,
-      children: (
-        <CreatePublishTarget
-          closeDialog={() => setAddDialogHidden(true)}
-          current={null}
-          targets={publishTargets || []}
-          types={publishTypes}
-          updateSettings={savePublishTarget}
-        />
-      ),
     });
+    setDialogChildren(
+      <CreatePublishTarget
+        closeDialog={() => setDialogHidden(true)}
```

Comment:
It's better to use a boolean state to control the visibility of `<CreatePublishTarget />` rather than save it to state.

Reply:
`<CreatePublishTarget />` is a reuse component for add new profile and edit current profile. So, it's not only the problem with visibility.  Can refactor this part later.